oommfc/oommf/oommf.py

```diff
@@ -299,150 +343,173 @@ def errors(self):
+        ``False`` will cause it to check for available methods again. Defaults
+        to ``True``.
 
-    docker_exe : str
+        envvar : str
 
-      The name or path of the docker command. Defaults to ``'docker'``.
+        Name of the environment variable containing the path to ``oommf.tcl``.
+        Defaults to ``'OOMMFTCL'``.
 
-    Returns
-    -------
-    oommfc.oommf.OOMMFRunner
+        oommf_exe : str
 
-        An OOMMF runner.
+        The name or path of the executable ``oommf`` command. Defaults to
+        ``'oommf'``.
 
-    Raises
-    ------
-    EnvironmentError
+        docker_exe : str
 
-        If no OOMMF can be found on host.
+        The name or path of the docker command. Defaults to ``'docker'``.
 
-    Examples
-    --------
-    1. Getting OOMMF Runner.
+        Returns
+        -------
+        oommfc.oommf.OOMMFRunner
 
-    >>> import oommfc as oc
-    ...
-    >>> runner = oc.oommf.get_oommf_runner()
-    >>> isinstance(runner, oc.oommf.OOMMFRunner)
-    True
+            An OOMMF runner.
 
-    """
-    global _cached_oommf_runner
-    if use_cache and (_cached_oommf_runner is not None):
-        return _cached_oommf_runner
-
-    # Check for the OOMMFTCL environment variable pointing to oommf.tcl.
-    oommf_tcl = os.environ.get(envvar, None)
-    if oommf_tcl is not None:
-        cmd = ['tclsh', oommf_tcl, 'boxsi',
-               '+fg', '+version', '-exitondone', '1']
+        Raises
+        ------
+        EnvironmentError
+
+            If no OOMMF can be found on host.
+
+        Examples
+        --------
+        1. Getting OOMMF Runner.
+
+        >>> import oommfc as oc
+        ...
+        >>> runner = oc.oommf.get_oommf_runner()
+        >>> isinstance(runner, oc.oommf.OOMMFRunner)
+        True
+
+        """
+        log.debug(f"Starting get_oommf_runner(use_cache={self.cache_runner}, "
+                  f"envvar={self.envvar}, oommf_exe={self.oommf_exe}, "
+                  f"docker_exe={self.docker_exe})")
+
+        # Check for the OOMMFTCL environment variable pointing to oommf.tcl.
+        log.debug(f"Step 1: Checking for the '{self.envvar=}' environment "
+                  "variable pointing to oommf.tcl.")
+        oommf_tcl = os.environ.get(self.envvar, None)
+        if oommf_tcl is not None:
+            cmd = [
+                'tclsh', oommf_tcl, 'boxsi', '+fg', '+version', '-exitondone',
+                '1'
+            ]
+            try:
+                res = sp.run(cmd, stdout=sp.PIPE, stderr=sp.PIPE)
+            except FileNotFoundError:
+                log.warning('oommf.tcl was not found.')
+            else:
+                if res.returncode:
+                    log.warning(
+                        'OOMMFTCL is set, but OOMMF could not be run.\n'
+                        f'stdout:\n{res.stdout}\nstderr:\n{res.stderr}')
+                else:
+                    self._runner = TclOOMMFRunner(oommf_tcl)
+                    return
```

Comment:
Documentation above says the method would return a runner object. The code here and below seems not to return an object (other than None).

Reply:
Outdated documentation, see above.